build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts

```diff
@@ -525,7 +531,14 @@ export abstract class LeafWithDoneFileTask extends LeafTask {
 	 * Subclass should override these to configure the leaf with done file task
 	 */
 
-	// The content to be written in the done file.
+	/**
+	 * The content to be written in the "done file".
+	 * @remarks
+	 * This file must have different content if the work needed to be done by this task changes.
+	 * This is typically done by listing and/or hashing the inputs to this task.
```

Comment:
```suggestion
	 * This is typically done by listing and/or hashing the inputs and outputs to this task.
```

Reply:
Assuming our tasks are pure functions (which I think our build assumes?) why do we need to test include the outputs? Is that to just guard against someone deleting or modifying the outputs without also deleting the done file? I'd like to include more details in this comment since the implications and motivations for including the outputs isn't obvious.

Also the policy check done file does not seem to include its outputs only its inputs: is that a bug?